packages/expo-image-picker/ios/MediaHandler.swift

```diff
@@ -182,21 +182,40 @@ internal struct MediaHandler {
               let videoUrl = url as? URL else {
           return completion(.failure(FailedToReadVideoException().causedBy(error)))
         }
-
-        let targetUrl = try generateUrl(withFileExtension: ".mov")
-        try VideoUtils.tryCopyingVideo(at: videoUrl, to: targetUrl)
-
-        guard let size = VideoUtils.readSizeFrom(url: targetUrl) else {
-          return completion(.failure(FailedToReadVideoSizeException()))
+        
+        // In case of passthrough, we want original file extension, mp4 otherwise
+        // TODO: (barthap) Support other file extensions?
+        let transcodeFileType = AVFileType.mp4
+        let transcodeFileExtension = ".mp4"
+        let originalExtension = ".\(videoUrl.pathExtension)"
+
+        // We need to copy the result into a place that we control, because the picker
+        // can remove the original file during conversion.
+        // Also, the transcoding may need a separate url - one of these will be used as a final result
+        let assetUrl = try generateUrl(withFileExtension: originalExtension)
+        let transcodedUrl = try generateUrl(withFileExtension: transcodeFileExtension)
+        try VideoUtils.tryCopyingVideo(at: videoUrl, to: assetUrl)
+        
+        VideoUtils.transcodeVideoAsync(sourceAssetUrl: assetUrl,
+                                       destinationUrl: transcodedUrl,
+                                       outputFileType: transcodeFileType,
+                                       exportPreset: options.videoExportPreset) { result in
+          if case .failure(let exception) = result {
+            return completion(.failure(exception))
+          }
+          let targetUrl = try! result.get()
```

Comment:
How to handle this better? I want to have exception inside clause and URL below. Tried that:
```swift
guard case let .success(targetUrl) = result else {
  return completion(.failure(/* How to get error here ??? */)
}
// can use targetUrl here
```
but cannot get error

Reply:
DO you still struggle with this one? 🤔

Reply:
Yes, possible solutions:

<details>
<summary>Option 1</summary>

```swift
switch result {
case .failure(let error):
  return completion(.failure(error))
case .success(let targetUrl):
  everythingBelowAsASeparateFunction(targetUrl)
}
```
</details>


<details>
<summary>Option 2</summary>

```swift
extension Result {
  public var failure: Failure? {
    switch self {
    case .failure(let error):
      return error
    default:
      return nil
    }
  }
}
```

and then

```swift
guard case let .success(targetUrl) = result else {
  return completion(.failure(result.failure!)
}
// can use targetUrl here
```

</details>

cc @tsapeta

Reply:
I would use the first one 😉